cmd/infra/powervs/service_id.go

```diff
@@ -188,6 +204,23 @@ func deleteServiceID(name, APIKey, accountID, resourceGroupID, crYaml, secretRef
 	return nil
 }
 
+func updateCRYaml(crYaml, templateName string, serviceInstanceValue string) string {
+	params := PolicyParams{
+		CloudInstanceID: serviceInstanceValue,
+	}
+	tmpl, err := template.New(templateName).Parse(crYaml)
+
+	if err != nil {
+		panic(fmt.Sprintf("programmer error, %s failed to parse: %v", templateName, err))
```

Comment:
No need to panic, you can return the error

Reply:
added return, thanks